File: frontend-react/src/types/resDetail.ts
export interface ResDetail {
  name: string;
  image_urls: string[];
  image_menu: string[];
  phone_number: string;
  price_range: string;
  opening_time?: string[] | string | null;
  description?: string;
  parking_info?: string;
  utilities?: number[];
  regulations?: string;
  requires_deposit?: boolean;
  deposit_amount?: number;
  deposit_min_guests?: number;
}

export interface RestaurantMenuItem {
  id: number;
  restaurant_id: number;
  name: string;
  category: string;
  price: number;
  description?: string | null;
  image_url?: string | null;
  is_available: boolean;
}

File: backend-python/schemas/resDetail.py
from typing import List, Optional
from pydantic import BaseModel, Field

class RestaurantDetailCreate(BaseModel):
    restaurant_id: int 
    image_urls: Optional[List[str]] = None
    image_menu: Optional[List[str]] = None
    price_range: Optional[str] = None
    phone_number: Optional[str] = None
    description: Optional[str] = None
    opening_time: Optional[List[str]] = None
    parking_info: Optional[str] = None
    utilities: Optional[List[int]] = None
    regulations: Optional[str] = None